### PR TITLE
feat: [#41] Expose questionnaire details from pogues

### PR DESCRIPTION
**Pull request by davdarras**
Expose questionnaire details extracted from Pogues
- retrieve questionnzitr from Pogues (JSON)
- extract and expose details from JSON
  - label
  - modes
- changing name variables to distinguish questionnaire id (questionnaire id from public enemy DB) and pogues id (questionnaire id from pogues)

### CODE REVIEW

File: src/main/java/fr/insee/publicenemy/api/infrastructure/questionnaire/entity/BaseQuestionnaireEntity.java [hunk 1]
```diff
@@ -91,7 +91,7 @@ public Questionnaire toModel() {
 
     public static BaseQuestionnaireEntity createFromModel(String campaignLabel, Questionnaire questionnaire) {
```
Comment (FBibonne) [inline src/main/java/fr/insee/publicenemy/api/infrastructure/questionnaire/entity/BaseQuestionnaireEntity.java:92]: check questionnaire is not null

File: src/main/java/fr/insee/publicenemy/api/infrastructure/questionnaire/entity/QuestionnaireEntity.java [hunk 2]
```diff
@@ -40,14 +40,18 @@ public QuestionnaireEntity(String questionnaireId, CampaignEntity campaign, Stri
 
     public static QuestionnaireEntity createFromModel(String campaignLabel, Questionnaire questionnaire) {
```
Comment (FBibonne) [inline src/main/java/fr/insee/publicenemy/api/infrastructure/questionnaire/entity/QuestionnaireEntity.java:41]: check questionnaire is not null

File: src/main/java/fr/insee/publicenemy/api/infrastructure/questionnaire/entity/QuestionnaireEntity.java [hunk 3]
```diff
@@ -40,14 +40,18 @@ public QuestionnaireEntity(String questionnaireId, CampaignEntity campaign, Stri
...
         questionnaireEntity.setCreationDate(date);
         questionnaireEntity.setUpdatedDate(date);
         return questionnaireEntity;
     }
 
+    /**
+     * Update questionnaire entity from questionnaire
+     * @param questionnaire
+     */
     public void update(Questionnaire questionnaire) {
```
Comment (FBibonne) [inline src/main/java/fr/insee/publicenemy/api/infrastructure/questionnaire/entity/QuestionnaireEntity.java:55]: check questionnaire is not null

**Comment by davdarras** [comment]
> * Use record for class Questionnaire too

not sure about this one, constructors will become ugly no ?

> 
> * In DdiPoguesServiceImpl :
>   
>   * in .getJsonPogues  exception raised if questionnaireId is invalid : the exceptions is never caught : need a 404 ?
>   * many places where method arguments are supposed to be not null :  annotate with @nonnull (getJsonPogues, getXmlDdi ...)
>   * make sure tha returned object are nerver null or use optionals
>   * getXmlDdi : wath should happen is there is an exception during WS call
> 
> * Service classes should depend on Dao interfaces instead of implementations (ex SurveyItemsServices)
> 
> * no method of DDIUseCase seems to be transactional : remove @transactional ?

done

> 
> * use of FileNameAwareByteArrayResource instead of ByteArrayResource is really mandatory ?

Won't work with ByteArrayResource unless i give filename, not really elegant in this way but it works

> 
> * **ENABLE Sonar analyzes**

Got problem wih sonarcloud, waiting for the project to be created, but sonarlint is now enabled and many little fixes have been done

> 
> * **UNIT TESTS**
Will come with next features

> 
> * Add spring boot plugin spring-boot-maven-plugin to build an executable jar or a war ?

It's already included in feature/ci branch, will come in next PR

**Comment by FBibonne** [comment]
About records, you can keep the existing constructors. You just have to "plug them " with the default one (all args constructor)

**Review by FBibonne — COMMENTED** [review]
- One another possible record : ApiError
- Why ApiExceptionHandler is @Order ?
- Try mutualize methods in ApiExceptionHandler : many exceptions have the same treatment
- FilenameAwareByteArrayResource.equals shouldn't override parent class definition cf [liskov substitution principle](https://gitlab.insee.fr/craftsmanship/documentation/-/wikis/concepts/SOLID#-liskov-substitution-principle-)
-

**Comment by davdarras** [comment]
> * One another possible record : ApiError
> 
> 
>     * Try mutualize methods in ApiExceptionHandler : many exceptions have the same treatment
> 
>     * FilenameAwareByteArrayResource.equals shouldn't override parent class definition cf [liskov substitution principle](https://gitlab.insee.fr/craftsmanship/documentation/-/wikis/concepts/SOLID#-liskov-substitution-principle-)

Done

>     * Why ApiExceptionHandler is @order ?

leftover from a previous test, got rid of it :)

**Review by FBibonne — APPROVED** [review]

**Pull request merged**